### PR TITLE
Replace goerli with sepolia as the testnet

### PR DESCRIPTION
**Pull request by portdeveloper**
Replaces goerli with sepolia throughout the app. Kept everything in mev-boost and did as little changes as possible to keep this PR small.

## This PR:
-> removes relay-goerli.ts and implements a mechanism to detect the chain id in relay.ts
-> Removes unused chains from networks.ts
-> Implements a new function getNetworkConfig in networks.ts that returns relevant flashbots-related urls
-> Uses JsonRpcProvider instead of AlchemyProvider in useRecoveryProcess as sepolia is not supported for AlchemyProvider in the version of ethers we are using
-> Changes how the asset detection logic entirely: Uses getTokenBalances, getTokenMetadata, and getNftsForOwner instead of getAssetTransfers, as that endpoint does not support sepolia testnet.
-> Updates the readme and the relevant pieces of code to say "sepolia" instead of "goerli"

### Next steps
- [ ] Think about testing? (maybe playwright?)
- [ ] Might want to start using MEV-Share (It works on a hwr clone i built, so the code is ready)

### CODE REVIEW

**Comment by escottalexander** [comment]
Love to see all this refactoring. I will test extensively and leave a review when I get a chance. Thanks, @portdeveloper!

File: packages/nextjs/pages/api/relay.ts [hunk 1]
```diff
@@ -5,16 +5,28 @@ import {
...
-);
 
 export default async function handler(req: any, res: any) {
+  // Get network from request headers or query params
+  const networkId = req.headers["x-network-id"] || req.query.networkId;
+  const network = Number(networkId) === chains.sepolia.id ? chains.sepolia : chains.mainnet;
+  const networkConfig = getNetworkConfig(network);
+
+  const provider = new ethers.providers.JsonRpcProvider(
+    `https://eth-sepolia.g.alchemy.com/v2/${process.env.NEXT_PUBLIC_ALCHEMY_API_KEY}`,
```
Comment (escottalexander) [inline packages/nextjs/pages/api/relay.ts:20]: This is hardcoded to the sepolia Alchemy endpoint.  I guess it should use `networkConfig.alchemyUrl`.

**Review by escottalexander — CHANGES_REQUESTED** [review]
One small change need to be made. Everything looks so great! The way you are getting the assets is so SO much better. It also shows assets that the old method never showed in my wallet. Nice work on this.

File: packages/nextjs/pages/api/relay.ts (hunk 1 above)
Comment (portdeveloper) [reply packages/nextjs/pages/api/relay.ts:20]: Oh nice catch! I should've looked better for this one!

**Review by escottalexander — APPROVED** [review]
LGTM!

**Pull request merged**